### PR TITLE
Filtered topics are considered invalid

### LINKED ISSUE #177: Filtered topics are considered invalid
If you hide topics with  "kafka.explorer.topics.filter", they're considered invalid:

Screenshot 2021-04-30 at 15 21 09

They should not be proposed for completion, since they're hidden, but should still be considered valid.

### PR DESCRIPTION
**Pull request by angelozerr**
Filtered topics are considered invalid

Fixes #177

Signed-off-by: azerr <azerr@redhat.com>

### CODE REVIEW

File: src/client/client.ts [hunk 1]
```diff
@@ -471,3 +473,37 @@ export function addQueryParameter(query: string, name: string, value?: string):
     }
     return `${query}${query.length > 0 ? '&' : '?'}${name}=${value}`;
 }
+
+export function isShowTopic(t: Topic): boolean {
```
Comment (fbricon) [inline src/client/client.ts:477]: isVisible(t: Topic)

File: src/kafka-file/kafkaFileClient.ts [hunk 2]
```diff
@@ -9,44 +9,46 @@ import { ConsumerLaunchStateProvider, getLanguageService, LanguageService, Produ
...
-        catch (e) {
-            this.topics = [];
+        if (!this.filteredTopics) {
+            try {
+                this.allTopics = await this.loadTopics();
+            }
+            catch (e) {
+                this.allTopics = [];
+            }
+            const topics = this.allTopics
```
Comment (fbricon) [inline src/kafka-file/kafkaFileClient.ts:34]: const visibleTopics

File: src/client/client.ts (hunk 1 above)
Comment (angelozerr) [reply src/client/client.ts]: fixed

File: src/kafka-file/kafkaFileClient.ts (hunk 2 above)
Comment (angelozerr) [reply src/kafka-file/kafkaFileClient.ts]: fixed

**Comment by fbricon** [comment]
thanks @angelozerr

**Pull request merged**